src/binary/uint.rs

```diff
@@ -73,6 +69,45 @@ impl UInt {
     }
 }
 
+/// The result of the [`magnitude_to_bytes`] function. The fields are private;
+/// users must go through `as_ref()` to see the slice that ignores leading
+/// zeros.
+///
+/// This type exists to avoid heap allocation.
+pub struct U64Magnitude {
+    magnitude_bytes: [u8; mem::size_of::<u64>()],
+    first_occupied_byte: usize,
+}
+
+impl AsRef<[u8]> for U64Magnitude {
+    fn as_ref(&self) -> &[u8] {
+        &self.magnitude_bytes[self.first_occupied_byte..]
+    }
+}
+
+/// Returns the magnitude as big-endian bytes.
```

Comment:
I would just say:

```suggestion
/// Converts a `u64` to a [`U64Magnitude`.]
```

The rework of the doc comment makes me wonder if it would be more idiomatic as just `impl From<u64> for U64Magnitude`.

Reply:
I love the convert traits, but I think this one is too unintuitive. Zack suggested `encode_uint`, which I like, so I went with that.